accounts-db/src/accounts_db.rs

```diff
@@ -7663,15 +7667,17 @@ impl AccountsDb {
         {
             let chunk_size = std::cmp::max(1, len / quarter_thread_count()); // # pubkeys/thread
             let batches = 1 + len / chunk_size;
-            (0..batches)
-                .into_par_iter()
-                .map(|batch| {
-                    let start = batch * chunk_size;
-                    let end = std::cmp::min(start + chunk_size, len);
-                    update(start, end)
-                })
-                .flatten()
-                .collect::<Vec<_>>()
+            self.thread_pool.install(|| {
```

Comment:
Background process such as shrinking may call this function. So we don't want to use foreground thread_pool here.

Reply:
Maybe a better way is  to make this function accept threadpool reference as argument so caller has to decide?

Reply:
```
-- a. forground (store_unfrozen)

1. TX foreground process always pass "Inline". so it will not use threads.
store_cached_inline (tx processing)
    - no thread

2. individual account store on bank.store_account()
    - no thread. only one account at a time.

3. bulk accounts store on bank bank.store_accounts, e.g. epoch rewards, rents
(no more)
    - may use thread 

-- b. background (store_frozen)
    - store ancient --- we can use "clean_pool" too.
    - shrink ---- in this case, upper in the call chain, `shrink_candidate_slots()`, we already use thread_pool_clean.
    - flush cache -- we can use "clean_pool" too.


So we can use "clean_pool" for (b) and for a.3 we want to use a different
pool. 
```

Reply:
In the current code it would appear there are only 2 paths to update_index, one that sets threading policy to single thread, and the one from store_frozen. Based on this, I've set the function to default to thread_pool_clean.

Reply:
I think we don't want `a.3` to end up using `thread_pool_clean`. `a.3` is on the critical path for foreground processing.